src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -437,6 +516,51 @@ private static boolean isExprSurrounded(DetailAST ast) {
         return ast.getFirstChild().getType() == TokenTypes.LPAREN;
     }
 
+    /**
+     * Checks whether an expression is surrounded by parentheses.
+     *
+     * @param ast the {@code DetailAST} to check if it is surrounded by
+     *        parentheses.
+     */
+    private void checkExpression(DetailAST ast) {
+        // If 'parentToSkip' == 'ast', then we've already logged a
+        // warning about an immediate child node in visitToken, so we don't
+        // need to log another one here.
+        if (parentToSkip != ast && isExprSurrounded(ast)) {
+            if (assignDepth >= 1) {
+                log(ast, MSG_ASSIGN);
+            }
+            else if (ast.getParent().getType() == TokenTypes.LITERAL_RETURN) {
+                log(ast, MSG_RETURN);
+            }
+            else {
+                log(ast, MSG_EXPR);
+            }
+        }
+
+        parentToSkip = null;
+    }
+
+    /**
+     * Checks if conditional, relational, unary and postfix operators
+     * in expressions are surrounded by parentheses.
+     *
+     * @param ast the {@code DetailAST} to check if it is surrounded by
+     *        parentheses.
+     * @return {@code true} if the expression is surrounded by
+     *         parentheses.
+     */
+    private static boolean checkAroundOperators(DetailAST ast) {
+        final int type = ast.getType();
+        final DetailAST parent = ast.getParent();
+        return (isInTokenList(type, CONDITIONALS)
+                    || isInTokenList(type, RELATIONAL)
```

Comment:
I think it is better to merge CONDITIONALS and RELATIONAL lists into one and simplify this checks. Same for lines below

Reply:
done.